src/actinia_tiling_plugin/api/merge/patch_merge.py

```diff
@@ -227,36 +233,104 @@ def _patch_vector(self, vect):
         self._execute_process_list(plv)
-        self.num_of_steps += self.num_steps["vector"]
+        self.num_of_steps += self.num_steps["vector"][0]
+        self._set_progress()
+
+    def _prepare_patch_strds(self):
+        """Prepares the patch of STRDS by reading one STRDS and adding the
+        rasters to the list of rasters to patch and writing other information
+        to a the dict self.strds_infos.
+        """
+        for strds in self.strds:
+            # strds get list of rasters and other STRDS infos
+            tpl_values_strds = {"strds": f"{strds}@{self.mapsetlist[0]}"}
+            pl_strds, pconv = pctpl_to_pl(
+                "patch/pc_strds_list_rasters.json", tpl_values_strds)
+            self.output_parser_list = pconv.output_parser_list
+            self._execute_process_list(pl_strds)
+            self._parse_module_outputs()
+            strds_rasters = self.module_results["rasters"]
+            strds_desc = self.module_results["strds_description"]
+            strds_info = self.module_results["strds_info"]
+            col_names = strds_rasters[0].split("|")
+            strds_raster_infos = dict()
+            strds_raster_infos["rasters"] = list()
+            for idx in range(1, len(strds_rasters)):
+                rast_info = dict()
+                rinfos = strds_rasters[idx].split("|")
+                for j in range(len(rinfos)):
+                    rast_info[col_names[j]] = rinfos[j]
+                rast_info["all"] = strds_rasters[idx]
+                strds_raster_infos["rasters"].append(rast_info)
+                if rast_info["name"] not in self.raster_maps:
+                    self.raster_maps.append(rast_info["name"])
+            desc = "".join([entry[2:] if entry not in [
+                "# Title:", "# Description:", "# Command history:"] else entry
+                for entry in strds_desc])
+            strds_raster_infos["title"] = re.findall(
+                r"# Title:(.*?)# Description:", desc
+            )[0]
+            strds_raster_infos["description"] = re.findall(
+                r"# Description:(.*?)# Command history:", desc
+            )[0]
+            strds_raster_infos["temporaltype"] = strds_info["temporal_type"]
+            strds_raster_infos["semantictype"] = strds_info["semantic_type"]
+            self.strds_infos[strds] = strds_raster_infos
+            self.num_of_steps += self.num_steps["strds"][0]
+            self.step += (
+                self.num_steps["raster"][0] * len(
+                    strds_raster_infos["rasters"]
+                )
+            )
+            self._set_progress()
+
+    def _patch_strds(self, strds_name, strds_info):
+        """Creates new STRDS as a duplicate with patched rasters.
+        """
+        id = str(uuid4())
+        tmp_strds_file = os.path.join(
+            global_config.TMP_WORKDIR, f"strds_{strds_name}_{id}.txt"
+        )
+        # write file to register the rasters in the STRDS without the mapset
```

Comment:
Why do you add rasters without the mapset? I think if no mapset is given, the current mapset is automatically used which might be wrong.

Reply:
No the current mapset is the right one, because I patched the rasters in the tmp mapsets to one raster in the current mapset.

Reply:
Great!